src/Kentico.Xperience.Lucene/Admin/UIPages/IndexListingPage.cs

```diff
@@ -73,39 +81,161 @@ public override Task ConfigurePage()
+
+        if (permissions.Rebuild)
+        {
+            PageConfiguration.TableActions.AddCommand(LocalizationService.GetString("Build index"), nameof(Rebuild), Icons.RotateRight);
+        }
+        if (permissions.Update)
+        {
+            PageConfiguration.TableActions.AddCommand("Edit", nameof(Edit));
+        }
+        if (permissions.Delete)
+        {
+            PageConfiguration.TableActions.AddCommand("Delete", nameof(Delete));
+        }
+        if (permissions.Create)
+        {
+            PageConfiguration.HeaderActions.AddLink<IndexCreatePage>("Create");
+        }
+
+        await base.ConfigurePage();
+    }
+
+    /// <inheritdoc/>
+    protected override async Task<LoadDataResult> LoadData(LoadDataSettings settings, CancellationToken cancellationToken)
+    {
+        var permissions = await GetUIPermissions();
+
+        try
+        {
+            var statistics = await luceneClient.GetStatistics(cancellationToken);
+
+            // Add statistics for indexes that are registered but not created in Lucene
+            AddMissingStatistics(ref statistics);
+
+            // Remove statistics for indexes that are not registered in this instance
+            var filteredStatistics = statistics.Where(stat =>
+                LuceneIndexStore.Instance.GetAllIndices().Any(index => index.IndexName.Equals(stat.Name, StringComparison.OrdinalIgnoreCase)));
+
+            var searchedStatistics = DoSearch(filteredStatistics, settings.SearchTerm);
+            var orderedStatistics = SortStatistics(searchedStatistics, settings);
+            var rows = orderedStatistics.Select(stat => GetRow(stat, permissions));
+
+            return new LoadDataResult
+            {
+                Rows = rows,
+                TotalCount = rows.Count()
+            };
+        }
+        catch (Exception ex)
+        {
+            EventLogService.LogException(nameof(IndexListingPage), nameof(LoadData), ex);
+            return new LoadDataResult
+            {
+                Rows = Enumerable.Empty<Row>(),
+                TotalCount = 0
+            };
+        }
+    }
+
+    private Row GetRow(LuceneIndexStatisticsViewModel statistics, UIPermissions permissions)
+    {
+        if (statistics.Name is null || LuceneIndexStore.Instance.GetIndex(statistics.Name) is not LuceneIndex index)
+        {
+            throw new InvalidOperationException($"Unable to retrieve Lucene index with name '{statistics.Name}.'");
+        }
+
+        var row = new Row
+        {
+            Identifier = index.Identifier,
+            Cells = new List<Cell>
+            {
+                new StringCell
+                {
+                    Value = statistics.Name
+                },
+                new StringCell
+                {
+                    Value = statistics.Entries.ToString()
+                },
+            }
+        };
+
+        var actions = new List<Action>();
+
+        if (permissions.Update)
```

Comment:
Row columns are added based on permissions so that the rows match the table heading columns